### PR TITLE
Do not imply --marionette flag

### PR DESCRIPTION
**Pull request by andreastt**
Users of mozrunner may not want to imply starting the Marionette remote
control server when they start Firefox.  This removes the --marionette
flag from being inferred when building the command.

It also addresses the issue that Windows systems do not understand
double-dashed GNU style flags.  I will later submit a patch to geckodriver
which adds a single-dashed X11 style -marionette flag when calling
mozrunner.

Fixes: https://github.com/mozilla/geckodriver/issues/640

### CODE REVIEW

File: src/runner.rs [hunk 1]
```diff
@@ -112,9 +112,13 @@ impl Runner for FirefoxRunner {
...
 
     fn build_command(&self, command: &mut Command) {
-        command.args(&self.args[..]).arg("--marionette").arg("--profile").arg(&self.profile.path)
-            .stdout(Stdio::inherit()).stderr(Stdio::inherit())
-            .env("MOZ_NO_REMOTE", "1").env("NO_EM_RESTART", "1");
+        command
+            .env("MOZ_NO_REMOTE", "1")
+            .env("NO_EM_RESTART", "1")
+            .args(&self.args[..])
+            .arg("--profile").arg(&self.profile.path)
```
Comment (jgraham) [inline src/runner.rs]: This should be `-profile`, right?

**Review by jgraham — CHANGES_REQUESTED** [review]

**Comment by andreastt** [comment]
Filed https://github.com/mozilla/geckodriver/pull/641 for parity.

File: src/runner.rs (hunk 1 above)
Comment (andreastt) [reply src/runner.rs]: Yes it should. Pushed 06a56b68120c8a766923aa376d7cbebba80ad9b5 as a fixup.

**Pull request merged**